src/unix/thread.c

```diff
@@ -387,16 +387,19 @@ int uv_sem_init(uv_sem_t* sem, unsigned int value) {
   uv_sem_t semid;
   struct sembuf buf;
   int err;
+  union {
+    int val;
+    struct semid_ds *buf;
+    unsigned short *array;
```

Comment:
Fixed the star position.
Regarding semun, the z/OS header files don't declare that union. 😃 
So the program has to explicitly define and declare.